### PR TITLE
darkradiant: init at 3.9.0

### LINKED ISSUE #358707: Package request: darkradiant
## Project description

Open-source level editor for Doom 3 and The Dark Mod
## Metadata

* homepage URL: https://www.darkradiant.net/
* source URL: https://github.com/codereader/DarkRadiant
* license: [DarkRadiant License](https://github.com/codereader/DarkRadiant/blob/master/LICENSE)
* platforms: unix

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by theobori**
Fix #358707

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by keenanweaver** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358833`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>darkradiant</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
Working great. Thank you so much for packaging this so quickly!

File: pkgs/by-name/da/darkradiant/package.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      apple-sdk_11
+    ];
+
+  # Tests source files are using an undefined reference to the openGL library
+  doCheck = false;
+
+  cmakeFlags = [
+    # Disabling dynamic rpath, otherwise it will not found the needed libraries within $out/lib/darkradiant
+    "-DENABLE_RELOCATION=OFF"
+    "-DENABLE_DM_PLUGINS=${if buildPlugins then "ON" else "OFF"}"
```
Comment (pluiedev) [inline pkgs/by-name/da/darkradiant/package.nix:75]:
```suggestion
    (lib.cmakeBool "ENABLE_RELOCATION" false)
    (lib.cmakeBool "ENABLE_DM_PLUGINS" buildPlugins)
```

File: pkgs/by-name/da/darkradiant/package.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  ];
+
+  postPatch = ''
+    substituteInPlace radiantcore/CMakeLists.txt \
+      --replace-fail "\$ORIGIN/.." "$out/lib/darkradiant"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/share/man/man1
+    cp ../man/darkradiant.1 $out/share/man/man1/
```
Comment (pluiedev) [inline pkgs/by-name/da/darkradiant/package.nix:85]: Use `installManPage`

**Review by pluiedev — CHANGES_REQUESTED** [review]
Just some nitpicks

**Review by pluiedev — APPROVED** [review]
LGTM & TYSM!

**Comment by FliegendeWurst** [comment]
Build fails on Darwin, please mark as broken or fix.
```
-- Checking for module 'gl'
--   No package 'gl' found
CMake Error at /nix/store/6paqqjczjfv8z2i6qr7gdfq21n0rbglr-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:645 (message):
  The following required packages were not found:
   - gl
Call Stack (most recent call first):
  /nix/store/6paqqjczjfv8z2i6qr7gdfq21n0rbglr-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:873 (_pkg_check_modules_internal)
  CMakeLists.txt:56 (pkg_check_modules)
-- Configuring incomplete, errors occurred!
```

**Comment by theobori** [comment]
> Build fails on Darwin, please mark as broken or fix.
> 
> ```
> -- Checking for module 'gl'
> --   No package 'gl' found
> CMake Error at /nix/store/6paqqjczjfv8z2i6qr7gdfq21n0rbglr-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:645 (message):
>   The following required packages were not found:
>    - gl
> Call Stack (most recent call first):
>   /nix/store/6paqqjczjfv8z2i6qr7gdfq21n0rbglr-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:873 (_pkg_check_modules_internal)
>   CMakeLists.txt:56 (pkg_check_modules)
> -- Configuring incomplete, errors occurred!
> ```

For the time being, I will indicate that the package is broken for darwin, I am not able to test on mac.

File: pkgs/by-name/da/darkradiant/package.nix [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    repo = "DarkRadiant";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-y0VzTnHobW36/25/nTV49OKnUMpnsjImioMdNKoTyYA=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    asciidoctor
+    wrapGAppsHook3
```
Comment (FliegendeWurst) [inline pkgs/by-name/da/darkradiant/package.nix:44]:
```suggestion
    wrapGAppsHook3
    wxGTK32
```

File: pkgs/by-name/da/darkradiant/package.nix [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      libgit2
+      python3
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [ libX11 ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_11
+    ];
+
+  # Tests source files are using an undefined reference to the openGL library
+  doCheck = false;
```
Comment (FliegendeWurst) [inline pkgs/by-name/da/darkradiant/package.nix:72]:
```suggestion
  doCheck = true;
```
Works for me

**Pull request merged**